src/Command/Email/ConfigurePlatformEmailCommand.php

```diff
@@ -0,0 +1,279 @@
+use React\EventLoop\Loop;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Exception\ValidatorException;
+use Symfony\Component\Validator\Validation;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Class ConfigurePlatformEmailCommand.
+ */
+class ConfigurePlatformEmailCommand extends CommandBase {
+
+  protected static $defaultName = 'email:configure';
+
+  /**
+   * @var int
+   */
+  private $domain_verification_error;
+
+  /**
+   * {inheritdoc}.
+   */
+  protected function configure() {
+    $this->setDescription('Configure Platform email for one or more applications')
+      ->addArgument('subscriptionUuid', InputArgument::OPTIONAL)
+      ->addOption('domain', NULL, InputOption::VALUE_REQUIRED)
+      ->setAliases(['ec']);
+  }
+
+  /**
+   * @param \Symfony\Component\Console\Input\InputInterface $input
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   *
+   * @return int
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    $this->io->writeln('Welcome to Platform Email setup! This script will walk you through setting up Platform Email, all through the command line and using the Cloud API!');
+
+    $base_domain = $this->determineDomain($input);
+    $client = $this->cloudApiClientService->getClient();
+    $subscription = $this->determineCloudSubscription();
+    $response = $client->request('post', "/subscriptions/{$subscription->uuid}/domains", [
+      'form_params' => [
+        'domain' => $base_domain,
+      ],
+    ]);
+    // @todo Check response!
+
+    $domain_uuid = $this->fetchDomainUuid($client, $subscription, $base_domain);
+
+    $this->io->success([
+      "Great! You've registered the domain {$base_domain} to subscription {$subscription->name}.",
+      "We will create a text file with the DNS records for your newly registered domain",
+      "Provide these records to your DNS provider",
+      "After you've done this, please continue."
+    ]);
+    $file_format = $this->io->choice('Would you like your output in JSON or YAML format?', ['YAML', 'JSON'], 'YAML');
+    $this->createDnsText($client, $subscription, $domain_uuid, $file_format);
+    $continue = $this->io->confirm('Have you finished providing the DNS records to your DNS provider?');
+    if (!$continue) {
+      $this->io->info("Make sure to give these records to your DNS provider, then rerun this script with the domain that you just registered.");
+      return 1;
+    }
+
+    $this->pollDomainRegistrationsUntilSuccess($subscription, $domain_uuid, $this->output);
+
+    // Allow for as many reverification tries as needed.
+    while (isset($this->domain_verification_error)) {
```

Comment:
The way that this property is set and used doesn't pass the sniff test for me. It seems like pollDomainRegistrationsUntilSuccess() should return a boolean to indicate success or failure.

Reply:
I think I'd then have something like this: 
```
while (!this->pollDomainRegistrationsUntilSuccess($subscription, $domain_uuid, $this->output)) {
        $retry_verification = $this->io->confirm('Would you like to retry verification?');
      if (!$retry_verification) {
	         $this->io->writeln('Please check your DNS records with your DNS provider and try again by rerunning this script with the domain that you just registered.');
        return 1;
      }
}
```